client/client_pool/src/concord_client_pool.cpp

```diff
@@ -506,18 +506,21 @@ void ConcordClientPool::processReplies(ClientPtr &client,
   ClientPoolMetrics_.average_req_dur_gauge.Get().Set((uint64_t)average_req_dur_.avg());
   if (average_req_dur_.numOfElements() == 1000) average_req_dur_.reset();  // reset the average every 1000 samples
   ClientPoolMetrics_.executed_requests_counter++;
-  metricsComponent_.UpdateAggregator();
-  auto finish = std::chrono::steady_clock::now();
-  for (const auto &reply : replies.second) {
-    auto before_send = client->getAndDeleteCidBeforeSendTime(reply.cid);
-    auto arrival_time = cid_arrival_map_[reply.cid];
-    cid_arrival_map_.erase(reply.cid);
-    duration = std::chrono::duration_cast<std::chrono::milliseconds>(before_send - arrival_time).count();
-    average_cid_receive_dur_.add(duration);
-
-    auto after_send = client->getAndDeleteCidResponseTime(reply.cid);
-    duration = std::chrono::duration_cast<std::chrono::milliseconds>(finish - after_send).count();
-    average_cid_close_dur_.add(duration);
+  {
+    std::unique_lock<std::mutex> lock(clients_queue_lock_);
+    metricsComponent_.UpdateAggregator();
```

Comment:
Why does this need to be under a lock?

Reply:
I agree, I will move it out of the lock scope. Metrics have their own sync mechanism.